source/melpomene/src/main.rs

```diff
@@ -153,77 +152,110 @@ fn kernel_entry(opts: MelpomeneOptions) {
+                    x: 400,
+                    y: PROFONT_12_POINT.underline.offset as i32,
+                },
+            )
+            .into_styled(line_style)
+            .draw(&mut fc_0)
+            .unwrap();
+            disp_hdl.draw_framechunk(fc_0).await.unwrap();
+        }
 
         k.spawn(
             async move {
-                'running: loop {
-                    // disp.clear(Gray8::BLACK).unwrap();
-
-                    tline.draw(&mut fc_0).unwrap();
-                    bline.draw(&mut fc_0).unwrap();
-                    rline.draw(&mut fc_0).unwrap();
-                    lline.draw(&mut fc_0).unwrap();
-
-                    text1.draw(&mut fc_0).unwrap();
-                    text2.draw(&mut fc_0).unwrap();
-
-                    let time = Local::now();
-
-                    let time_str = format!(
-                        "{:02}:{:02}:{02}",
-                        time.hour(),
-                        time.minute(),
-                        time.second()
-                    );
-
-                    let date_str =
-                        format!("{:02}/{:02}/{:02}", time.month(), time.day(), time.year());
-
-                    let date_text = Text::new(&date_str, Point::new(28, 35), datetime_style);
-                    let time_text = Text::new(&time_str, Point::new(88, 35), datetime_style);
-
-                    date_text.draw(&mut fc_0).unwrap();
-                    time_text.draw(&mut fc_0).unwrap();
-                    {
-                        let raw_img = fc_0.frame_display().unwrap();
-                        let image = Image::new(&raw_img, Point::new(80, 45));
-                        image.draw(&mut sdisp).unwrap();
-
-                        window.update(&sdisp);
-                        if window.events().any(|e| e == SimulatorEvent::Quit) {
-                            break 'running;
-                        }
+                let style = ring_drawer::BwStyle {
+                    background: Gray8::BLACK,
+                    font: MonoTextStyle::new(&PROFONT_12_POINT, Gray8::WHITE),
+                };
+
+                // At 12-pt font, there is enough room for 16 lines, with 50 chars/line.
+                //
+                // Leave out 4 for the implicit margin of two characters on each gutter.
+                let mut rline = RingLine::<16, 46>::new();
 
-                        Delay::new(Duration::from_secs(1)).await;
-                        sdisp.clear(Gray8::BLACK).unwrap();
+                loop {
+                    // Wait until there is a frame buffer ready. There wouldn't be if we've spammed frames
+                    // before they've been consumed.
+                    let mut fc_0 = loop {
+                        let fc = disp_hdl
+                            .get_framechunk(0, char_y as i32, 400, 240 - char_y)
+                            .await;
+                        if let Some(fc) = fc {
+                            break fc;
```

Comment:
i was wondering if this could be expressed nicer as a `while` loop, but i guess it would require a `mut Option<FrameChunk>` outside of the loop, so 🤷‍♀️